packages/react/menu/src/Menu.tsx

```diff
@@ -621,6 +636,206 @@ const MenuItemIndicator = React.forwardRef((props, forwardedRef) => {
+  MenuSubMenuContentOwnProps
+>;
+
+const MenuSubMenuContent = React.forwardRef((props, forwardedRef) => {
+  const { ...contentProps } = props;
+  const context = useMenuContext(SUB_MENU_CONTENT_NAME);
+  const subMenuContext = useSubMenuContext(SUB_MENU_CONTENT_NAME);
+  const trigger = subMenuContext.triggerRef.current;
+
+  return (
+    <MenuContent
+      side="right"
+      align="start"
+      {...contentProps}
+      portalled
+      ref={forwardedRef}
+      onOpenAutoFocus={composeEventHandlers(contentProps.onOpenAutoFocus, (event) => {
+        const content = event.target as HTMLElement;
+        if (content && subMenuContext.focusFirstItem) {
+          const items = Array.from(content.querySelectorAll(ENABLED_ITEM_SELECTOR));
+          (items[0] as HTMLElement | undefined)?.focus();
+        }
+      })}
+      onKeyDown={composeEventHandlers(contentProps.onKeyDown, (event) => {
+        if (event.key === 'ArrowLeft') {
+          // Close a single level
+          event.stopPropagation();
+          context.onOpenChange(false);
+          trigger?.focus();
+        }
+      })}
+      onEscapeKeyDown={composeEventHandlers(contentProps.onEscapeKeyDown, () => trigger?.focus())}
+      /**
+       * Prevent focusing previous element on close.
+       * We don't want the focus to flicker back and forth between the content and trigger when mouse scrubbing.
+       */
+      onCloseAutoFocus={composeEventHandlers(contentProps.onCloseAutoFocus, (event) =>
+        event.preventDefault()
+      )}
+      /**
+       * Prevent needlessly dismissing the menu when clicking the trigger
+       * The menu should always stay open while the mouse is hovering
+       */
+      onPointerDownOutside={composeEventHandlers(contentProps.onPointerDownOutside, (event) => {
+        if (trigger?.contains(event.target as HTMLElement)) {
+          event.preventDefault();
+        }
+      })}
+    />
+  );
+}) as MenuSubMenuContentPrimitive;
+
+MenuSubMenuContent.displayName = SUB_MENU_CONTENT_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * MenuSubMenuTrigger
+ * -----------------------------------------------------------------------------------------------*/
+
+const SUB_MENU_TRIGGER_NAME = 'MenuSubMenuTrigger';
+
+type MenuSubMenuTriggerOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof MenuItem>,
+  Polymorphic.OwnProps<typeof MenuAnchor>
+>;
+type MenuSubMenuTriggerPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuItem>,
+  MenuSubMenuTriggerOwnProps
+>;
+
+const MenuSubMenuTrigger = React.forwardRef((props, forwardedRef) => {
+  const { disabled, ...triggerProps } = props;
+  const context = useMenuContext(SUB_MENU_TRIGGER_NAME);
+  const subMenuContext = useSubMenuContext(SUB_MENU_TRIGGER_NAME);
+  const [mouseInteracting, setMouseInteracting] = React.useState(false);
+
+  return (
+    <MenuAnchor as={Slot}>
+      <MenuItem
```

Comment:
Composing via `Slot` as it felt simpler, the TS event types also seem to get lost when using `as`. Looking at this now perhaps `SubMenuTrigger` should default to `button`?